### PR TITLE
Apply the fix scan duration & code refactoring to this branch

### PR DESCRIPTION
**Pull request by tdincer**

### CODE REVIEW

File: element_calcium_imaging/imaging.py [hunk 1]
```diff
@@ -264,6 +265,15 @@ def make(self, key):
                 ProcessingTask * ProcessingParamSet * ProcessingMethod * scan.Scan & key
             ).fetch1("processing_method")
 
+            image_files = (
+                ProcessingTask * scan.Scan * scan.ScanInfo * scan.ScanInfo.ScanFile
+                & key
+            ).fetch("file_path")
```
Comment (dimitri-yatsenko) [inline element_calcium_imaging/imaging.py:271]: Do you need to join the other tables?
```suggestion
            image_files = (scan.ScanInfo.ScanFile & key).fetch("file_path")
```

File: element_calcium_imaging/imaging_no_curation.py [hunk 2]
```diff
@@ -264,6 +265,15 @@ def make(self, key):
                 ProcessingTask * ProcessingParamSet * ProcessingMethod * scan.Scan & key
```
Comment (dimitri-yatsenko) [inline element_calcium_imaging/imaging_no_curation.py:265]: looks like unnecessary tables are joined here.

**Review by kabilar — APPROVED** [review]
Thanks @tdincer!

File: element_calcium_imaging/scan.py [hunk 3]
```diff
@@ -378,6 +362,32 @@ def make(self, key):
             nd2_file = nd2.ND2File(scan_filepaths[0])
             is_multiROI = False  # MultiROI to be implemented later
 
+            # Frame per second
+            try:
+                fps = 1000 / nd2_file.experiment[0].parameters.periods[0].periodDiff.avg
+            except:
```
Comment (dimitri-yatsenko) [inline element_calcium_imaging/scan.py:368]: Catch specific exception types.

**Review by dimitri-yatsenko — CHANGES_REQUESTED** [review]

File: element_calcium_imaging/scan.py [hunk 4]
```diff
@@ -378,6 +362,32 @@ def make(self, key):
...
+                ti = (
+                    nd2_scan_obj.frame_metadata(0)
+                    .channels[0]
+                    .time.absoluteJulianDayNumber
+                )  # Initial frame's JD.
+                tf = (
+                    nd2_scan_obj.frame_metadata(nd2_scan_obj.shape[0] - 1)
+                    .channels[0]
+                    .time.absoluteJulianDayNumber
+                )  # Final frame's JD.
```
[the remark this answers is not in the document]
Comment (tdincer) [reply element_calcium_imaging/scan.py:383]: JD: Julian Date

**Comment by tdincer** [comment]
@dimitri-yatsenko  I'll address your comments!

File: element_calcium_imaging/scan.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dimitri-yatsenko) [reply element_calcium_imaging/scan.py:383]: then it's better to say `Julian date` rather than JD

**Pull request merged**